internal/fusefrontend/xattr.go

```diff
@@ -124,21 +116,13 @@ func (fs *FS) ListXAttr(path string, context *fuse.Context) ([]string, fuse.Stat
 }
 
 // encryptXattrName transforms "user.foo" to "user.gocryptfs.a5sAd4XAa47f5as6dAf"
-func (fs *FS) encryptXattrName(attr string) (cAttr string, err error) {
-	// Reject anything that does not start with "user."
-	if !strings.HasPrefix(attr, xattrUserPrefix) {
-		return "", syscall.EPERM
-	}
+func (fs *FS) encryptXattrName(attr string) (cAttr string) {
 	// xattr names are encrypted like file names, but with a fixed IV.
 	cAttr = xattrStorePrefix + fs.nameTransform.EncryptName(attr, xattrNameIV)
-	return cAttr, nil
+	return cAttr
 }
 
 func (fs *FS) decryptXattrName(cAttr string) (attr string, err error) {
-	// Reject anything that does not start with "user.gocryptfs."
-	if !strings.HasPrefix(cAttr, xattrStorePrefix) {
-		return "", syscall.EINVAL
-	}
```

Comment:
I think we want to keep this

Reply:
Thanks! I returned it back, but probably this part can be also optimised even more, there is a code here https://github.com/rfjakob/gocryptfs/blob/master/internal/fusefrontend/xattr.go#L112 , which guarantees, that the method will not be called with a wrong parameter. What do you think, what is the best to prevent duplication of these checks?

Reply:
Good point. I'm traveling right now, will think about it and merge in the next days.